tests/libvmi/vmi.go

```diff
@@ -88,9 +88,10 @@ func WithRng() Option {
 // WithResourceMemory specifies the vmi memory resource.
 func WithResourceMemory(value string) Option {
 	return func(vmi *v1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
-			k8sv1.ResourceMemory: resource.MustParse(value),
+		if vmi.Spec.Domain.Resources.Requests == nil {
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{}
 		}
+		vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(value)
```

Comment:
IMO what should be done here is to add memory to existing memory if already defined. Something like:
```go
if vmi.Spec.Domain.Resources.Requests == nil {
	vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{}
}

memoryToDefine := resource.MustParse(value)
existingMemory, exists := vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory]

if exists {
	// This syntax is wrong, but you get the idea
	memoryToDefine += existingMemory
}

vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = memoryToDefine
```

But maybe this can be done in a follow-up PR

Reply:
For tests, I believe explicitly setting a value is more appropriate than increasing it.
So I would not recommend adding such logic, which is also not in sync with the name of this option.

Remember that the options are part of the creation of a VMI spec, not an update of it.
As a caller to a VMI spec creation, I would have to calculate what the value should be and not add it in small pieces.

Reply:
> For tests, I believe explicitly setting a value is more appropriate than increasing it.

I have a strong problem with that statement as I think it's a mistake to tightly couple libvmi with functional tests. IMHO libvmi shouldn't use any testing functions and reside outside of `tests/` folder.

> Remember that the options are part of the creation of a VMI spec, not an update of it.

I thought that we may have a scenario in which we're using `WithX` and `WithY` while X demands some allocated memory, and Y needs some more memory, hence the idea.

But in another thought - we allocate resource in `renderLaunchManifest` according to the VMI's spec. IOW, this is already done behind the scenes. In the current design, the resources that are defined on the VMI spec are solely dedicated to the guest. So I agree, it shouldn't be accumulated :)

Reply:
Agreed Thanks for reviewing!